specification.md

```diff
@@ -0,0 +1,89 @@
+
+OpenTelemetry and Prometheus/OpenMetrics have different design
+goals and this reflects in the data model and implementation
+details. This section summarizes a few key differences.
+
+* **Pull vs push**: Prometheus is mainly designed for pull
+  whereas OpenTelemetry primarily is designed for push. This
+  difference causes how the state is maintained throughout the
+  collection including how it’s handled in the OpenTelemetry
+  collector.
+* **Cumulative vs delta**: OpenTelemetry supports delta temporality
+  whereas Prometheus always expects absolute/cumulative values. This
+  breaks some components that produce and communicate deltas where
+  cumulatives cannot be rebuilt before being exported to Prometheus.
+* **Histogram boundaries**: Prometheus histogram boundaries are by
+  lower equal (le) while OpenTelemetry histogram boundaries are
+  greater equal (ge). (This difference will be resolved via #18)
+* **Semantic conventions**: OpenTelemetry predefines semantic
+  conventions to collect additional metadata with telemetry data.
+  Prometheus users don’t follow the same conventions and Prometheus
+  client library provided data may lack the semantic conventions
+  available in OpenTelemetry libraries.
+
+## Compatibility Requirements
+
+Given the number of fundamental design goals between OpenTelemetry
+and Prometheus, our aim is to close the gaps where possible and
+make the right compromises to meet the goals defined in this document.
+OpenTelemery and Prometheus won’t be fully compatible but we will
+enable important use cases to enable OpenTelemetry for Prometheus
+users. The following sections summarizes the expectations from
+the collector and the libraries.
+
+### Collector
+
+* Collector will implement a Prometheus remote write exporter.
+  Publishing a pull-based metrics handler with all collected
+  metrics is not a scalable approach.
+* Collector will support scraping and ingesting cumulative metrics.
+  Prometheus doesn’t support deltas and there are cases where
+  rebuilding the cumulatives from deltas is not possible/easy.
+* Collector will support all discovery and scraping configuration
+  options in the Prometheus server. Collector will ignore the
+  alerting rules.
+* Collector will support exporting cumulative series to
+  OLTP-compatible exporters.
+* Collector will produce an “up” metric as a gauge that is compatible
+  with the Prometheus server “up” metric. If scrape succeeds,
+  it will produce 1 and 0 if it fails.
+* Collector will produce "instance" and "job" labels similar
+  to the Prometheus server.
+* If a target disappears from the scrape, the collector will
+  write explicit staleness markers for the respective timeseries.
+* Collector won’t assume any OpenTelemetry semantic conventions
+  might be in place in the scraped data. Collector may decorate
+  the samples with some semantic convention attributes available
+  in the collector.
+* Collector will support WAL if it has access to persistent storage.
```

Comment:
Do we consider the WAL an external API, or just an internal implementation detail?  I agree it solves a number of problems, but i'm not sure if it is required to be compatible with prometheus

Reply:
If you're talking about Prometheus WAL, that would make remote write support easier to implement.

Reply:
This was about Prometheus WAL for the purposes of being able to reuse the Prometheus remote write. Today, we don't reuse the remote write from Prometheus and maintenance is a burden.